contracts/modules/v2/Assessment.sol

```diff
@@ -0,0 +1,276 @@
+
+  Claim[] public override claims;
+  address[] public override claimants;
+
+  Incident[] public override incidents;
+  mapping(uint104 => AffectedToken) internal tokenAffectedByIncident;
+
+  /* ========== CONSTRUCTOR ========== */
+
+  constructor (address masterAddress) {
+    // [todo] Move to intiialize function
+    // The minimum cover premium is 2.6%. 20% of the cover premium is: 2.6% * 20% = 0.52%
+    config.rewardPercentage = 52; // 0.52%
+    config.incidentExpectedPayoutPercentage = 3000; // 30%
+    config.claimAssessmentDepositPercentage = 500; // 5% i.e. 0.05 ETH submission flat fee
+    config.minVotingPeriodDays = 3; // days
+    config.maxVotingPeriodDays = 30; // days
+    config.payoutCooldownDays = 1; //days
+    master = INXMMaster(masterAddress);
+    nxm = INXMToken(master.tokenAddress());
+  }
+
+  /* ========== VIEWS ========== */
+
+  function tokenController() internal view returns (ITokenController) {
+    return ITokenController(getInternalContractAddress(ID.TC));
+  }
+
+  function memberRoles() internal view returns (IMemberRoles) {
+    return IMemberRoles(getInternalContractAddress(ID.MR));
+  }
+
+  function pool() internal view returns (IPool) {
+    return IPool(getInternalContractAddress(ID.P1));
+  }
+
+  function cover() internal view returns (ICover) {
+    return ICover(getInternalContractAddress(ID.CO));
+  }
+
+  function getVoteCountOfAssessor(address assessor) external override view returns (uint) {
+    return votesOf[assessor].length;
+  }
+
+  function getClaimsCount() external override view returns (uint) {
+    return claims.length;
+  }
+
+  function getIncidentsCount() external override view returns (uint) {
+    return incidents.length;
+  }
+
+  /* === MUTATIVE FUNCTIONS ==== */
+
+  /**
+   *  Submits a claim for assessment
+   *
+   *  @dev This function requires an ETH submission fee. See: _getSubmissionFee()
+   *
+   *  @param coverId          Cover identifier
+   *  @param requestedAmount  The amount expected to be received at payout
+   *  @param withProof        When true, a ProofSubmitted event is emitted with ipfsProofHash.
+   *                          When false, no ProofSubmitted event is emitted to save gas if the
+   *                          cover wording doesn't enforce a proof of loss.
+   *  @param ipfsProofHash    The IPFS hash required for proof of loss. It is ignored if withProof
+   *                          is false
+   */
+  function submitClaim(
+    uint24 coverId,
+    uint96 requestedAmount,
+    bool withProof,
+    string calldata ipfsProofHash
+  ) external payable override onlyMember {
+    AssessmentClaimsLib.submitClaim(
+      config,
+      internalContracts,
+      claims,
+      claimants,
+      coverId,
+      requestedAmount,
+      withProof,
+      ipfsProofHash
+    );
+
+  }
+
+  function submitIncident(
+    uint24 productId,
+    uint96 priceBefore,
+    uint32 date
+  ) external override {
```

Comment:
this should only be callable by an AB member

Reply:
I see this is checked in the library, but it's inconsistent with the submitClaim function above. pick one approach and stick to it

Reply:
also, why not have a similar submitIncident function in the library, as you did for the claims? let's keep it consistent